apps/src/templates/sectionProgress/summary/VirtualizedSummaryView.jsx

```diff
@@ -30,17 +27,26 @@ class VirtualizedSummaryView extends Component {
     section: sectionDataPropType.isRequired,
     scriptData: scriptDataPropType.isRequired,
     lessonOfInterest: PropTypes.number.isRequired,
-    getLevels: PropTypes.func,
+    levelsByLesson: PropTypes.object,
     onScroll: PropTypes.func,
     jumpToLessonDetails: PropTypes.func.isRequired
   };
 
-  state = {
-    fixedColumnCount: 1,
-    fixedRowCount: 1,
-    scrollToColumn: 0,
-    scrollToRow: 0
-  };
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
NIT: personally i think it's cleaner to declare state outside of the constructor. looking through our code, that is much more common than doing it in the construction. major caveat to that observation though is that none of the places we do it that way include a constructor, so maybe that's why you changed it here?

i'd love to conform on this and i'm fine with the idea of "put it in the constructor if you're using a constructor" if you're strongly opinionated about that. however, i think there's value in being consistent and not doing it differently when there is a constructor. it also means less code change when adding or removing a constructor (as in this case).

however, that begs the question of where to declare other instance properties (e.g. `summaryView`). my personal vote would be to only put things in the constructor that need to reference `this` (e.g. bindings) to avoid using constructors at all unless necessary (just since so few of our components use them, so seems more consistent and imo cleaner). 

but idk, maybe we should just roll with personal preference instead of trying to conform on this kind of thing? not sure how useful it is for us to conform to conventions that are specific to our cabal for things that aren't specified in the org-wide style guide.

Reply:
I don't have a specific reason for doing this. I just noticed that everywhere we use a constructor, we declare state in the constructor (Caveat: This may just be left over from earlier versions of JS). Additionally, the react docs for constructors also declared state in the constructor: https://reactjs.org/docs/react-component.html#constructor. I don't have too strong of a preference, but I'm all for making things consistent. Rather than deciding as a cabal, I'd prefer to decide as an engineering team. There's enough shared code between teams that these kinds of decisions should be made at an org level and added to the style guide.